iap/src/main/java/com/example/iap/VerifyIapRequestHeader.java

```diff
@@ -97,37 +55,15 @@ boolean verifyJwtForComputeEngine(
   }
 
   private boolean verifyJwt(String jwtToken, String expectedAudience) throws Exception {
-
-    // parse signed token into header / claims
-    SignedJWT signedJwt = SignedJWT.parse(jwtToken);
-    JWSHeader jwsHeader = signedJwt.getHeader();
-
-    // header must have algorithm("alg") and "kid"
-    Preconditions.checkNotNull(jwsHeader.getAlgorithm());
-    Preconditions.checkNotNull(jwsHeader.getKeyID());
-
-    JWTClaimsSet claims = signedJwt.getJWTClaimsSet();
-
-    // claims must have audience, issuer
-    Preconditions.checkArgument(claims.getAudience().contains(expectedAudience));
-    Preconditions.checkArgument(claims.getIssuer().equals(IAP_ISSUER_URL));
-
-    // claim must have issued at time in the past
-    Date currentTime = Date.from(Instant.now(clock));
-    Preconditions.checkArgument(claims.getIssueTime().before(currentTime));
-    // claim must have expiration time in the future
-    Preconditions.checkArgument(claims.getExpirationTime().after(currentTime));
-
-    // must have subject, email
-    Preconditions.checkNotNull(claims.getSubject());
-    Preconditions.checkNotNull(claims.getClaim("email"));
-
-    // verify using public key : lookup with key id, algorithm name provided
-    ECPublicKey publicKey = getKey(jwsHeader.getKeyID(), jwsHeader.getAlgorithm().getName());
-
-    Preconditions.checkNotNull(publicKey);
-    JWSVerifier jwsVerifier = new ECDSAVerifier(publicKey);
-    return signedJwt.verify(jwsVerifier);
+    TokenVerifier tokenVerifier = TokenVerifier.newBuilder()
+        .setAudience(expectedAudience)
+        .build();
+    try {
+      tokenVerifier.verify(jwtToken);
```

Comment:
What is the recommended way to verify the claims (e.g. pull the `email` and `sub` field from the verified JWT)? It would be good to include this.